### PR TITLE
Graduate sysctls to GA 

### PR DESCRIPTION
**Pull request by wgahnagl**
#### What type of PR is this?

/ kind feature

#### What this PR does / why we need it:
graduates sysctls to GA 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
sysctls is now GA and locked to default
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/34-sysctl-fields#graduation  
```

### CODE REVIEW

**Review by ehashman — APPROVED** [review]
/lgtm

/hold
I think I'm a feature approver now, want to give node a chance to take a look

/assign @derekwaynecarr

**Comment by ehashman** [comment]
xref https://github.com/kubernetes/enhancements/issues/34

**Comment by ehashman** [comment]
/lgtm cancel

@wgahnagl looks like there are some tests in the `pull-kubernetes-bazel-test` job that will need to be updated and/or removed now that we are locking this to true, can you PTAL?

File: pkg/features/kube_features.go
Comment (pacoxu) [inline pkg/features/kube_features.go:135]: The current release would be 1.21.

Comment (wgahnagl) [reply pkg/features/kube_features.go]: fixed !

**Comment by pacoxu** [comment]
as @ehashman mentioned， you need to check test failures and fix them. 
So the bazel test can pass.

After it is locked to true， some test cases can be removed.

File: pkg/api/pod/util_test.go
Comment (ehashman) [inline pkg/api/pod/util_test.go:1180]: This can be simplified out I think, we can just set enabled = true and remove the loop.

**Comment by ehashman** [comment]
Hmm not sure why the bazel job is still failing with the test removed...

Can you rebase?

**Comment by pacoxu** [comment]
> Hmm not sure why the bazel job is still failing with the test removed...
> 
> Can you rebase?

pkg/api/podsecuritypolicy/util_test.go:
  191: func TestDropSysctls(t *testing.T) {

There is another TestDropSysctls that should be removed as well.
@wgahnagl  would you remove this case as well.

File: pkg/features/kube_features.go
Comment (pacoxu) [inline pkg/features/kube_features.go:133]: sysctl feature is implemented in 1.4
but there is no feature gate, it use an alpha annotation.

IMO, we need not add this line.

**Comment by pacoxu** [comment]
@wgahnagl I complete the PR in #99263 base on this one and add you as co-author in the commit.

Feel free to close my pr if you want to work on this.

This’s conflict which need rebase here.

File: pkg/features/kube_features.go
Comment (ehashman) [reply pkg/features/kube_features.go:133]: It's useful for tracking the implementation history. I don't think this comment does any harm.

**Comment by ehashman** [comment]
This LGTM, the only real difference from #99263 is that PR removes the feature gates.

I don't understand why that wouldn't happen in the release where the feature gates actually get removed; otherwise, if for some reason we want to turn it back off the reverts would be very messy. What is the point of going through the step to lock the feature gate to true and not just remove it entirely if we're going to remove all the codepaths in advance of removal? cc @dims

**Review by ehashman — APPROVED** [review]
/lgtm

File: pkg/features/kube_features.go
Comment (pacoxu) [reply pkg/features/kube_features.go:133]: agree. no harm with it.

**Review by pacoxu — APPROVED** [review]

**Comment by pacoxu** [comment]
A small comment on PR description part, additional documentation for KEP link should be like https://github.com/kubernetes/kubernetes/blame/master/.github/PULL_REQUEST_TEMPLATE.md#L66
```
- [KEP]:  link
```

**Comment by liggitt** [comment]
/approve

for removal of locked GA feature gate checking in API helpers.

**Comment by liggitt** [comment]
> What is the point of going through the step to lock the feature gate to true and not just remove it entirely if we're going to remove all the codepaths in advance of removal?

Feature gates are user-visible in the CLI invocation. Locking to true at GA lets us simplify the code while giving deployments that opted into the feature early and are explicitly enabling it time to adjust their invocations to drop the explicit enable.

https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-feature-or-behavior tried to describe this (but I acknowledge it is confusing):

> Feature gates are intended to cover the development life cycle of a feature - they are not intended to be long-term APIs. As such, they are expected to be deprecated and removed after a feature becomes GA or is dropped.
> 
> As a feature moves through the stages, the associated feature gate evolves. The feature life cycle matched to its corresponding feature gate is:
> 
> * Alpha: the feature gate is disabled by default and can be enabled by the user.
> * Beta: the feature gate is enabled by default and can be disabled by the user.
> * GA: the feature gate is deprecated (see "Deprecation") and becomes non-operational.
> * GA, deprecation window complete: the feature gate is removed and calls to it are no longer accepted.

**Comment by ehashman** [comment]
/hold cancel

Thanks @liggitt ! @dims also put together https://github.com/kubernetes/community/pull/5545

**Pull request merged**
